graph/is_graph_bipartite.cpp

```diff
@@ -0,0 +1,142 @@
+/**
+ *  @file is_graph_bipartite
+ *
+ *	@brief Algorithm to check whether a graph is bipartite
+ *	(https://en.wikipedia.org/wiki/Bipartite_graph)
+ *
+ *	@details
+ *	A graph is a collection of nodes also called vertices and these vertices
+ *	are connected by edges.A bipartite graph is a graph whose vertices can be
+ * 	divided into two disjoint and independent sets U and V such that every edge 
+ *	connects a vertex in U to one in V. 	
+ *	
+ *	The given Algorithm will determine whether the given graph is bipartite or not
+ *
+ *
+ * 	Example - Here is a graph g1 with 5 vertices and is bipartite
+ *
+ *		1   4
+ *	   / \ / \
+ *	  2   3   5
+ *	
+ *	Example - Here is a graph G2 with 3 vertices and is not bipartite
+ *
+ *		1 --- 2
+ *		 \   /
+ *		   3
+ *
+ *	
+ *	@author [Akshat Vaya](https://github.com/AkVaya)
+ *		
+ */
+#include <iostream>
+#include <vector>
+#include <queue>
+
+const int nax = 5e5 + 1;
+/**
+ * Class for representing graph as an adjacency list.
+ */
+class graph {
+ 	private: 
+ 		int n;						/// size of the graph
+
+    	std::vector<std::vector <int> > adj;	/// adj stores the graph as an adjacency list
+
+    	std::vector<int> side;			///stores the side of the vertex
+
+ 	public:
+    /**
+     * @brief Constructor that initializes the graph on creation
+     */
+    graph(int size = nax){
+    	n = size;
+    	adj.resize(n);
+    	side.resize(n,-1);
+    }
+
+    void addEdge(int u, int v);     /// function to add edges to our graph
+
+	bool is_bipartite();    	/// function to check whether the graph is bipartite or not
+
+};
+/**
+ * @brief Function that add an edge between two nodes or vertices of graph
+ *
+ * @param u is a node or vertex of graph
+ * @param v is a node or vertex of graph
+ */
+void graph::addEdge(int u, int v) {
+    adj[u-1].push_back(v-1);
+    adj[v-1].push_back(u-1);
+}
+/**
+ *	@brief function that checks whether the graph is bipartite or not
+ *	the function returns true if the graph is a bipartite graph
+ *	the function returns false if the graph is not a bipartite graph
```

Comment:
Please mention how does this function check if the graph is bipartite. What is the name of the method used.

Reply:
👍 Thank you to @bhattah and @AkVaya